### PR TITLE
refactor: add disable_dashboard option and disable dashboard in metasrv and datanode

### LINKED ISSUE #1337: Disable dashboard in the http servers of datanode and metasrv
@MichaelScofield The `--http-addr` can resolve the port confliction, but it seems the `/dashboard` endpoint is still working:

```
$ ./greptime metasrv start --store-addr 127.0.0.1:2379 --server-addr 0.0.0.0:3002 --http-addr 0.0.0.0:14000

$ curl -XGET http://localhost:14000/dashboard/
```

IMO, we should:

1. Add the user configuration to determine whether the component should start with `/dashboard`;
2. When in distributed mode, the metasrv and datanode should disable `/dashboard`;

_Originally posted by @zyy17 in https://github.com/GreptimeTeam/greptimedb/issues/1334#issuecomment-1498434380_

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add disable_dashboard option and disable dashboard in metasrv and datanode.

Resolve #1337 

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by sunng87 — APPROVED** [review]

**Comment by sunng87** [comment]
wait, we dont need to provide option to disable it we just disable it by default because it's not usable with datanode and metasrv

**Comment by zyy17** [comment]
@sunng87 Your suggestion is right. I use `#[serde(skip)]` to [skip](https://github.com/GreptimeTeam/greptimedb/pull/1343/files#diff-2547faf5feae48c746017b1865ef203989efca274c6e58fa63bf814b325404e7R125) the `disable_dashboard` filed so it can't be parse from toml file. Only the frontend cli option can change `disable_dashboard` field.

**Comment by MichaelScofield** [comment]
> wait, we dont need to provide option to disable it we just disable it by default because it's not usable with datanode and metasrv

+1. It's better for user not need to explicitly disable Dashboard when starting Datanode or Metasrv.

File: src/cmd/src/frontend.rs [hunk 1]
```diff
@@ -149,18 +151,24 @@ impl TryFrom<StartCommand> for FrontendOptions {
...
-                ..Default::default()
-            });
+            if let Some(ref mut http_options) = opts.http_options {
+                http_options.addr = addr;
+            }
         }
+
+        if let Some(ref mut http_options) = opts.http_options {
+            http_options.disable_dashboard = cmd.disable_dashboard
+        }
```
Comment (shuiyisong) [inline src/cmd/src/frontend.rs:163]:
```suggestion
        // init http options
        let mut http_options = HttpOptions {
            disable_dashboard: cmd.disable_dashboard,
            ..Default::default()
        };
        if let Some(addr) = cmd.http_addr {
            http_options.addr = addr;
        }
        opts.http_options = Some(http_options);
```

How about simplify initialization a little bit? :)

**Review by shuiyisong — APPROVED** [review]

**Comment by zyy17** [comment]
> > wait, we dont need to provide option to disable it we just disable it by default because it's not usable with datanode and metasrv
> 
> +1. It's better for user not need to explicitly disable Dashboard when starting Datanode or Metasrv.

The datanode and metasrv will disable dashboard internal.

**Review by waynexia — APPROVED** [review]
LGTM

**Comment by paomian** [comment]
LGTM

**Review by paomian — APPROVED** [review]

**Pull request merged**